test/recipe.e2e-spec.ts

```diff
@@ -0,0 +1,234 @@
+          expect(response.body.fetchedRecipes).toHaveLength(3);
+        })
+
+        .expect(HttpStatus.OK);
+    });
+
+    it('should not fetch all recipes and return 401 error (UNAUTHORIZED)', () => {
+      return request(app.getHttpServer())
+        .get(`/recipes`)
+        .expect(HttpStatus.UNAUTHORIZED);
+    });
+  });
+
+  describe('/recipe/:id (PATCH)', () => {
+    const requestData = {
+      title: faker.word.noun(5),
+      description: faker.lorem.text(),
+      ingredients: faker.lorem.word(8),
+      preparation: faker.lorem.lines(4),
+    };
+    it('should update specific recipe by by given id', async () => {
+      const recipe = await prismaService.recipe.create({ data: recipeData });
+      return request(app.getHttpServer())
+        .patch(`/recipes/${recipe.id}`)
+        .send(requestData)
+        .set({ Authorization: `Bearer ${accessToken}` })
+        .expect((response: request.Response) => {
+          const { title, description, ingredients, preparation } =
+            response.body;
+          expect(title).toEqual(requestData.title);
+          expect(description).toEqual(requestData.description);
+          expect(ingredients).toEqual(requestData.ingredients);
+          expect(preparation).toEqual(requestData.preparation);
+        })
+        .expect(HttpStatus.OK);
+    });
+
+    it('should not update recipe and return 404 error (NOT FOUND)', async () => {
+      const recipe = await prismaService.recipe.create({ data: recipeData });
+      return request(app.getHttpServer())
+        .patch(`/recipes/${recipe.id + 1}`)
+        .send(requestData)
+        .set({ Authorization: `Bearer ${accessToken}` })
+        .expect(HttpStatus.NOT_FOUND);
+    });
+
+    it('should not update specific recipe and return 401 error (UNAUTHORIZED)', async () => {
+      const recipe = await prismaService.recipe.create({ data: recipeData });
+      return request(app.getHttpServer())
+        .patch(`/recipes/${recipe.id}`)
+        .send(requestData)
+        .expect(HttpStatus.UNAUTHORIZED);
+    });
+  });
+
+  describe('/recipes/:id (DELETE)', () => {
+    it('should delete specific recipe by by given id', async () => {
+      const recipe = await prismaService.recipe.create({ data: recipeData });
+      return request(app.getHttpServer())
+        .delete(`/recipes/${recipe.id}`)
+        .set({ Authorization: `Bearer ${accessToken}` })
+        .expect(HttpStatus.OK);
+    });
+
+    it('should not delete recipe and return 404 error (NOT FOUND)', async () => {
+      const recipe = await prismaService.recipe.create({ data: recipeData });
+      return request(app.getHttpServer())
+        .delete(`/recipes/${recipe.id + 1}`)
+        .set({ Authorization: `Bearer ${accessToken}` })
+        .expect(HttpStatus.NOT_FOUND);
+    });
+
+    it('should not delete specific recipe and return 401 error (UNAUTHORIZED)', async () => {
+      const recipe = await prismaService.recipe.create({ data: recipeData });
```

Comment:
as recipe is created in every test within this `describe` block, you can move recipe creation to `beforeEach` - of course within this describe block